### PR TITLE
Added Parellism section to NServiceBus tuning guidance

### PR DESCRIPTION
**Pull request by ramonsmits**
- [ ] Improve upgrade guide for RabbitMQ transport where concurrency behavior changed

### CODE REVIEW

File: nservicebus/operations/tuning.md [hunk 1]
```diff
@@ -34,6 +34,17 @@ Infrastructure monitoring should be set up for the environment that hosts the en
...
 > The concurrency set in the endpoint configuration defines the concurrency of each endpoint instance, and not the aggregate concurrency across all endpoint instances. For example, if the endpoint configuration sets the concurrency to 4 and the endpoint is scaled-out to 3 instances, the combined concurrency will be 12 and not 4.
 
+## Parellism
+
+The configure allowed concurrency limit is the limit that NServiceBus enforces but NServiceBus will not enforce parellism.
+
+> [!NOTE]
+> Behavior can very between transports and between different versions of the same transport.
+
+For example, when handlers that invoke synchrounous code only will block when the transport has a fully asynchronous design and handler will be processed sequentially.
```
Comment (johnsimons) [inline nservicebus/operations/tuning.md:44]:
```suggestion
```

**Review by johnsimons — CHANGES_REQUESTED** [review]

File: nservicebus/operations/tuning.md (hunk 1 above)
Comment (johnsimons) [reply nservicebus/operations/tuning.md:44]: I think we better off without it

**Comment by johnsimons** [comment]
@ramonsmits after spending a few hours on this, we decided to not over complicate it and keep it simple and link to existing documentation on the subject.
Thoughts?

**Review by johnsimons — APPROVED** [review]

**Comment by danielmarbach** [comment]
LGTM

**Pull request merged**